navigation/insilico_rotater.h

```diff
@@ -36,40 +39,52 @@ class InSilicoRotater
     template<class Func>
     void rotate(Func func)
     {
-        size_t i = 0;
+        // If we are starting scanning from a point other than zero
+        if(m_BeginRoll != 0) {
+            // Roll image and mask (if required)
+            rollImage(m_Image, m_BeginRoll);
+            if (!m_MaskImage.empty()) {
+                rollImage(m_MaskImage, m_BeginRoll);
+            }
+        }
+        
+        size_t i = m_BeginRoll;
         while (true) {
-            func(m_Image, m_MaskImage, i);
+            func(m_Image, m_MaskImage, i - m_BeginRoll);
 
             i += m_ScanStep;
-            if (i >= (size_t) m_Image.cols) {
+            if (i >= m_EndRoll) {
                 break;
             }
 
-            rollImage(m_Image);
+            rollImage(m_Image, m_ScanStep);
             if (!m_MaskImage.empty()) {
-                rollImage(m_MaskImage);
+                rollImage(m_MaskImage, m_ScanStep);
             }
         }
     }
 
     size_t max() const
     {
-        return static_cast<size_t>(m_Image.cols) / m_ScanStep;
+        return (m_EndRoll - m_BeginRoll) / m_ScanStep;
     }
```

Comment:
I feel like this function should be renamed to ``size()`` or ``numRotations()`` or something now, because ``max()`` kinda sounds like it should just return ``m_EndRoll``.

Reply:
Excellent point!